pkg/resource/deploy/source_eval.go

```diff
@@ -1765,7 +1776,40 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 	customTimeouts := opts.CustomTimeouts
 
-	additionalSecretOutputs := opts.GetAdditionalSecretOutputs()
+	// Schema-aware changes
+	//
+	// At this point we will consult the resource's schema to determine if any of the properties we'll send onward should
+	// be modified or extended. Presently this includes:
+	//
+	// * Including `Secret` properties in additional secret outputs.
+
+	additionalSecretOutputsSet := mapset.NewSet[string]()
+	additionalSecretOutputsSet.Append(opts.GetAdditionalSecretOutputs()...)
+
+	//var version *semver.Version
+	//if parsedVersion, err := semver.Parse(req.GetVersion()); err == nil {
+	//  version = &parsedVersion
+	//}
+
+	//if typeToken, err := tokens.ParseTypeToken(req.Type); err == nil {
+	//  packageName := typeToken.Package().String()
+	//  if pkgReference, err := rm.schemaLoader.LoadPackageReference(packageName, version); err == nil {
+	//    if resourceSchema, found, err := pkgReference.Resources().Get(req.Type); err == nil && found {
+	//      for _, outputProperty := range resourceSchema.Properties {
+	//        if outputProperty.Secret {
+	//          additionalSecretOutputsSet.Add(outputProperty.Name)
+	//        }
+	//      }
+	//    }
+	//  }
+	//}
```

Comment:
We are making the schema load bearing. Going between having and not having a schema will start to cause diffs on some providers, since `[secret] => "foo"` is a change that should show up in state.

I would be very careful to distinguish between tolerable errors (there is no schema for this resource, technically OK) and non-tolerable errors (we tried to load the schema, but failed to launch), correctly erroring for non-tolerable errors.

We should log for tolerable errors.

Reply:
A) This shouldn't actually cause diffs because SDK-gen should have been setting the same things secret anyway
B) A provider returning an empty schema is fine and we should tolerate that and just do nothing. I think anything else is an error, I don't expect any providers to error on this method or return invalid schemas.

Reply:
A) is true only as long as the generated SDK agrees with the provider. This may not hold now for Go, since they provider may be a different version then the SDK used to consume it.

B) Yes

Reply:
A) True, but I'd consider that already buggy and the diff would be more correct. I don't think we should try to avoid that.